mitmproxy/optmanager.py

```diff
@@ -594,21 +601,33 @@ def serialize(
     ruamel.yaml.YAML().dump(data, file)
 
 
-def save(opts: OptManager, path: str, defaults: bool = False) -> None:
+def save(opts: OptManager, path: Path | str, defaults: bool = False) -> None:
     """
     Save to path. If the destination file exists, modify it in-place.
 
     Raises OptionsError if the existing data is corrupt.
     """
-    path = os.path.expanduser(path)
-    if os.path.exists(path) and os.path.isfile(path):
-        with open(path, encoding="utf8") as f:
+    path = Path(path).expanduser()
+    if path.exists() and path.is_file():
+        with path.open(encoding="utf8") as f:
             try:
                 data = f.read()
             except UnicodeDecodeError as e:
                 raise exceptions.OptionsError(f"Error trying to modify {path}: {e}")
     else:
         data = ""
 
-    with open(path, "w", encoding="utf8") as f:
+    with path.open("w", encoding="utf8") as f:
         serialize(opts, f, data, defaults)
+
+
+def relative_path(script_path: Path | str, *, relative_to: Path | str) -> Path:
+    """
+    Make relative paths found in config files relative to said config file,
+    instead of relative to where the command is ran.
+    """
+    script_path = Path(script_path)
+    # Edge case when $HOME is not an absolute path
+    if script_path.expanduser() != script_path and not script_path.is_absolute():
+        script_path = script_path.expanduser().resolve()
```

Comment:
Do we need to call .resolve() here? Symlinks may change, and users may change them intentionally. :)
If .resolve() makes things significantly easier I'm happy to live with this constraint though.

Reply:
> won't they need to reload the options object thus re-resolving the new symlink ?

mitmproxy live-reloads scripts, so you don't need to reload options. :) I think `.absolute()` is more accurate here, so let's try to stick with that.